pirlib/backends/argo_batch.py

```diff
@@ -0,0 +1,364 @@
+            }
+
+            # Add the volume mount spec to the volume mount list.
+            volume_mounts.append(mount_spec)
+
+        # Add temporary field for defining dependencies.
+        if inp.source.node_id:
+            dependencies.append(argo_name(inp.source.node_id))
+
+    # Create the template dictionary.
+    template = {
+        "name": argo_name(name),
+        "container": {
+            "image": image,
+            "command": command,
+            "volumeMounts": volume_mounts,
+        },
+        "volumes": volumes,
+        "dependencies": dependencies,
+    }
+
+    return template
+
+
+def create_template_from_graph(
+    graph_outputs_encoded: str, graph: pirlib.pir.Graph
+) -> Dict[str, Any]:
+    """Generates an Argo template dictionary from the provided Graph.
+
+    :param graph_outputs_encoded: base64 encoding of the graph outputs.
+    :type graph_outputs_encoded: str
+    :param graph: A PIRlib Graph object.
+    :type graph: pirlib.pir.Graph
+    :return: An Argo template dict for the Graph object.
+    :rtype: Dict[str, Any]
+    """
+    name = graph.id
+    image = graph.nodes[0].entrypoints["main"].image
+    command = ["python", "-m", __name__, "graph", graph_outputs_encoded]
+    volumes = [create_nfs_volume_spec("node_outputs", "OUTPUT")]
+    volume_mounts = [
+        {"name": argo_name("node_outputs"), "mountPath": "/mnt/node_outputs"}
+    ]
+
+    for inp in graph.inputs:
+        inp_name = inp.id
+        inp_env_var = f"INPUT_{inp_name}"
+
+        inp_volume_spec = create_nfs_volume_spec(inp_name, inp_env_var, readonly=True)
+        volumes.append(inp_volume_spec)
+
+        inp_mount_spec = {
+            "name": argo_name(inp_name),
+            "mountPath": f"/mnt/graph_inputs/{inp_name}",
+        }
+        volume_mounts.append(inp_mount_spec)
+
+    if graph.outputs:
+        volumes.append(create_nfs_volume_spec("graph_outputs", "OUTPUT"))
+        volume_mounts.append(
+            {"name": argo_name("graph_outputs"), "mountPath": "/mnt/graph_outputs"}
+        )
+
+    template = {
+        "name": argo_name(name),
+        "container": {
+            "image": image,
+            "command": command,
+            "volumeMounts": volume_mounts,
+        },
+        "volumes": volumes,
+    }
+
+    return template
+
+
+def argo_refactor(workflow_yaml: str) -> str:
+    """Applies Argo specific refactoring on the
+    generated YAML file string.
+
+    :param workflow_yaml: The YAML string defining an Argo workflow.
+    :type workflow_yaml: str
+    :return: Argo compatiable YAML string.
+    :rtype: str
+    """
+    workflow_yaml = workflow_yaml.replace("true", "yes")
+    workflow_yaml = workflow_yaml.replace("false", "no")
+
+    return workflow_yaml
+
+
+class ArgoBatchBackend(Backend):
+    def execute_parser(self) -> Optional[argparse.ArgumentParser]:
+        pass
+
+    def execute(self) -> Optional[argparse.ArgumentParser]:
+        pass
```

Comment:
execute function, we will need later but do we need execute_parser and generate_parser ?

Reply:
They need to be implemented as they are defined in the base class.